modules/builtin/src/backend/prompts/pattern.ts

```diff
@@ -0,0 +1,50 @@
+import { ExtractionResult, IO, Prompt, PromptConfig, ValidationResult } from 'botpress/sdk'
+
+import commonFields from './common'
+
+class PromptPattern implements Prompt {
+  private _regexPattern: string
+
+  constructor({ regexPattern }) {
+    this._regexPattern = regexPattern
+  }
+
+  extraction(event: IO.IncomingEvent): ExtractionResult | undefined {
+    const text = event.payload.text
+    if (text) {
+      return {
+        value: text,
+        confidence: 1
+      }
+    }
+  }
+
+  async validate(value): Promise<ValidationResult> {
+    if (value == undefined) {
+      return { valid: false, message: 'Provided value is invalid' }
+    }
+
+    if (!new RegExp(this._regexPattern).test(value)) {
+      return { valid: false, message: 'Value does not match regex pattern' }
```

Comment:
I think there should be some way to provide a format example or customize the message: You must provide a phone number in the format 555-555-5555 or "Please provide a valid phone number". Not sure a lot of users know about regex patterns